<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Code and data for “Assessing the quality of comparative genomics data and results with the cogeqc R/Bioconductor package” - 4&nbsp; Assessing orthogroup inference for Brassicaceae genomes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/05_synteny_assessment_fabaceae.html" rel="next">
<link href="../chapters/03_orthogroup_inference_dbs.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/04_orthogroup_inference_brassicaceae.html"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Assessing orthogroup inference for Brassicaceae genomes</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Code and data for “Assessing the quality of comparative genomics data and results with the cogeqc R/Bioconductor package”</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/01_busco_assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Assessing the completeness of Chlorophyta genomes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/02_orthogroup_assessment_orthobench.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Orthogroup assessment in Orthobench</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/03_orthogroup_inference_dbs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Assessing orthogroup inference in public databases</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/04_orthogroup_inference_brassicaceae.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Assessing orthogroup inference for Brassicaceae genomes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/05_synteny_assessment_fabaceae.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Assessing synteny detection in Fabaceae</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Appendices</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#orthogroup-inference" id="toc-orthogroup-inference" class="nav-link active" data-scroll-target="#orthogroup-inference"><span class="header-section-number">4.1</span> Orthogroup inference</a></li>
  <li><a href="#exploratory-analysis-of-orthogroup-inference-results" id="toc-exploratory-analysis-of-orthogroup-inference-results" class="nav-link" data-scroll-target="#exploratory-analysis-of-orthogroup-inference-results"><span class="header-section-number">4.2</span> Exploratory analysis of orthogroup inference results</a></li>
  <li><a href="#orthogroup-assessment" id="toc-orthogroup-assessment" class="nav-link" data-scroll-target="#orthogroup-assessment"><span class="header-section-number">4.3</span> Orthogroup assessment</a></li>
  <li><a href="#comparing-and-visualizing-homogeneity-statistics" id="toc-comparing-and-visualizing-homogeneity-statistics" class="nav-link" data-scroll-target="#comparing-and-visualizing-homogeneity-statistics"><span class="header-section-number">4.4</span> Comparing and visualizing homogeneity statistics</a>
  <ul class="collapse">
  <li><a href="#global-distributions" id="toc-global-distributions" class="nav-link" data-scroll-target="#global-distributions"><span class="header-section-number">4.4.1</span> Global distributions</a></li>
  <li><a href="#the-effect-of-species-choice" id="toc-the-effect-of-species-choice" class="nav-link" data-scroll-target="#the-effect-of-species-choice"><span class="header-section-number">4.4.2</span> The effect of species choice</a></li>
  <li><a href="#the-effect-of-mcl-inflation-parameters" id="toc-the-effect-of-mcl-inflation-parameters" class="nav-link" data-scroll-target="#the-effect-of-mcl-inflation-parameters"><span class="header-section-number">4.4.3</span> The effect of mcl inflation parameters</a></li>
  <li><a href="#the-effect-of-diamond-mode-default-vs-ultra" id="toc-the-effect-of-diamond-mode-default-vs-ultra" class="nav-link" data-scroll-target="#the-effect-of-diamond-mode-default-vs-ultra"><span class="header-section-number">4.4.4</span> The effect of DIAMOND mode (default vs ultra)</a></li>
  </ul></li>
  <li><a href="#functional-analysis-of-homogeneous-and-heterogeneous-gene-families" id="toc-functional-analysis-of-homogeneous-and-heterogeneous-gene-families" class="nav-link" data-scroll-target="#functional-analysis-of-homogeneous-and-heterogeneous-gene-families"><span class="header-section-number">4.5</span> Functional analysis of homogeneous and heterogeneous gene families</a></li>
  <li><a href="#is-there-an-association-between-og-score-and-og-gene-length" id="toc-is-there-an-association-between-og-score-and-og-gene-length" class="nav-link" data-scroll-target="#is-there-an-association-between-og-score-and-og-gene-length"><span class="header-section-number">4.6</span> Is there an association between OG score and OG gene length?</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session info</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Assessing orthogroup inference for Brassicaceae genomes</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Here, we will compare the protein domain-based approach in <code>cogeqc</code> to assess the impact of multiple combinations of parameters in OrthoFinder <span class="citation" data-cites="emms2019orthofinder">(<a href="#ref-emms2019orthofinder" role="doc-biblioref">Emms and Kelly 2019</a>)</span> in the accuracy of orthogroup inference. The data set used here will be a collection of Brassicaceae genomes. The parameters we will change are:</p>
<ol type="1">
<li>Program (<code>-S</code> option)</li>
</ol>
<ul>
<li>DIAMOND</li>
<li>DIAMOND ultrasensitive</li>
</ul>
<ol start="2" type="1">
<li>MCL inflation parameter (<code>-I</code>)</li>
</ol>
<ul>
<li>1</li>
<li>1.5 (default)</li>
<li>2</li>
<li>3</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cogeqc)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstatix)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"code"</span>, <span class="st">"utils.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="orthogroup-inference" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="orthogroup-inference"><span class="header-section-number">4.1</span> Orthogroup inference</h2>
<p>To start, we will load the proteome data and export each proteome as a FASTA file in the <code>data</code> directory, so we can pass it to OrthoFinder.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load proteomes</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"brassicaceae_proteomes.rda"</span>))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Write files to data/</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="fu">seq_along</span>(brassicaceae_proteomes), <span class="cf">function</span>(x) {</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    outfile <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="fu">paste0</span>(<span class="fu">names</span>(brassicaceae_proteomes)[x], <span class="st">".fasta"</span>))</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    Biostrings<span class="sc">::</span><span class="fu">writeXStringSet</span>(</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>        brassicaceae_proteomes[[x]], outfile</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we can run OrthoFinder for each combination of parameters. Here, we created 2 different bash scripts for each DIAMOND mode. They are:</p>
<ul>
<li><code>of_diamond.sh</code>: code to run DIAMOND (default mode) for different inflation parameters;</li>
<li><code>of_diamond_ultra.sh</code>: code to run DIAMOND in ultrasensitive mode for different inflation parameters</li>
</ul>
<p>The 2 files can be run with:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> of_diamond.sh</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> of_diamond_ultra.sh</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <em>Orthogroups.tsv</em> files were all moved to the directory <code>products/result_files</code>.</p>
</section>
<section id="exploratory-analysis-of-orthogroup-inference-results" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="exploratory-analysis-of-orthogroup-inference-results"><span class="header-section-number">4.2</span> Exploratory analysis of orthogroup inference results</h2>
<p>Now that we have the <em>Orthogroups.tsv</em> files from OrthoFinder, let’s load them.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract tar.xz file</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>tarfile <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"result_files"</span>, <span class="st">"Orthogroups.tar.xz"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>outdir <span class="ot">&lt;-</span> <span class="fu">tempdir</span>()</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(<span class="st">"tar"</span>, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"-xf"</span>, tarfile, <span class="st">"--directory"</span>, outdir))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get path to OrthoFinder output</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>og_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">path =</span> outdir, </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pattern =</span> <span class="st">"Orthogroups.*"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Read and parse files</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>ogs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(og_files, <span class="cf">function</span>(x) {</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    og <span class="ot">&lt;-</span> <span class="fu">read_orthogroups</span>(x)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    og <span class="ot">&lt;-</span> og <span class="sc">%&gt;%</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">Species =</span> stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(Species, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">Gene =</span> <span class="fu">str_replace_all</span>(</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>            Gene, <span class="fu">c</span>(</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>                <span class="st">"</span><span class="sc">\\</span><span class="st">.[0-9]$"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>                <span class="st">"</span><span class="sc">\\</span><span class="st">.[0-9]</span><span class="sc">\\</span><span class="st">.p$"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>                <span class="st">"</span><span class="sc">\\</span><span class="st">.t[0-9]$"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>                <span class="st">"</span><span class="sc">\\</span><span class="st">.g$"</span> <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(og)</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>og_names <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.tsv"</span>, <span class="st">""</span>, <span class="fu">basename</span>(og_files))</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>og_names <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Orthogroups_"</span>, <span class="st">""</span>, og_names)</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ogs) <span class="ot">&lt;-</span> og_names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s explore OG sizes for each combination of parameters and filter orthogroups by size to remove orthogroups that are artificially large.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize OG sizes</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>og_sizes_plot <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>default_1) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Default, mcl = 1"</span>), </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>default_1_5) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Default, mcl = 1.5"</span>) <span class="sc">+</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()),</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>default_2) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Default, mcl = 2"</span>) <span class="sc">+</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()), </span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>default_3) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Default, mcl = 3"</span>) <span class="sc">+</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>ultra_1) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Ultra, mcl = 1"</span>) <span class="sc">+</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()), </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>ultra_1_5) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Ultra, mcl = 1.5"</span>) <span class="sc">+</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()),</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>ultra_2) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Ultra, mcl = 2"</span>) <span class="sc">+</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()), </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_og_sizes</span>(ogs<span class="sc">$</span>ultra_3) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Ultra, mcl = 3"</span>) <span class="sc">+</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()),</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">ncol =</span> <span class="dv">8</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>og_sizes_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-fig.dpi="200">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../products/plots/og_sizes_plot.png" class="img-fluid figure-img" width="2550"></p>
<figcaption class="figure-caption">Orthogroup sizes for each run.</figcaption>
</figure>
</div>
</div>
</div>
<p>Expectedly, OrthoFinder runs with mcl inflation parameters of 1 lead to very large orthogroups, including some orthogroups with thousands of genes.</p>
<p>Now, let’s explore the percentage of orthogroups with &gt;200, &gt;100, and &gt;50 genes in each OrthoFinder run.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate OG sizes for each run</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>og_sizes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(ogs, <span class="cf">function</span>(x) {</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    sizes <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">table</span>(x<span class="sc">$</span>Orthogroup, x<span class="sc">$</span>Species))</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    total <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(sizes)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    sizes_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">unclass</span>(sizes))</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>    sizes_df<span class="sc">$</span>Total <span class="ot">&lt;-</span> total</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(sizes_df)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the percentage of OGs with &gt;=100 genes? And with &gt;50 genes?</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>percentage_size <span class="ot">&lt;-</span> <span class="cf">function</span>(size_df, <span class="at">n =</span> <span class="dv">100</span>) {</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sum</span>(size_df<span class="sc">$</span>Total <span class="sc">&gt;</span> n) <span class="sc">/</span> <span class="fu">nrow</span>(size_df) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>percentages <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">Mode =</span> <span class="fu">names</span>(og_sizes),</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">P200 =</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(og_sizes, percentage_size, <span class="at">n =</span> <span class="dv">200</span>)),</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">P100 =</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(og_sizes, percentage_size, <span class="at">n =</span> <span class="dv">100</span>)),</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">P50 =</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(og_sizes, percentage_size, <span class="at">n =</span> <span class="dv">50</span>)),</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">OGs =</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(og_sizes, nrow))</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder rows from lowest to highest mcl inflation</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>orders <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">"default_1"</span>, <span class="st">"default_1_5"</span>, <span class="st">"default_2"</span>, <span class="st">"default_3"</span>,</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ultra_1"</span>, <span class="st">"ultra_1_5"</span>, <span class="st">"ultra_2"</span>, <span class="st">"ultra_3"</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>percentages <span class="ot">&lt;-</span> percentages[orders, ]</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Visual exploration</span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>percentage_plot <span class="ot">&lt;-</span> percentages <span class="sc">%&gt;%</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">!</span>Mode) <span class="sc">%&gt;%</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">str_replace_all</span>(</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>        name,</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">c</span>(</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>            <span class="st">"OGs"</span> <span class="ot">=</span> <span class="st">"Number of OGs"</span>,</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>            <span class="st">"P200"</span> <span class="ot">=</span> <span class="st">"% OGs with &gt;200 genes"</span>,</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>            <span class="st">"P100"</span> <span class="ot">=</span> <span class="st">"% OGs with &gt;100 genes"</span>,</span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>            <span class="st">"P50"</span> <span class="ot">=</span> <span class="st">"% OGs with &gt;50 genes"</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">y =</span> Mode, <span class="at">x =</span> value)) <span class="sc">+</span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Mode), <span class="at">show.legend =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"ultra_3"</span> <span class="ot">=</span> <span class="st">"#08519C"</span>, <span class="st">"ultra_2"</span> <span class="ot">=</span> <span class="st">"#3182BD"</span>,</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"ultra_1_5"</span> <span class="ot">=</span> <span class="st">"#6BAED6"</span>, <span class="st">"ultra_1"</span> <span class="ot">=</span> <span class="st">"#BDD7E7"</span>,</span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_3"</span> <span class="ot">=</span> <span class="st">"#006D2C"</span>, <span class="st">"default_2"</span> <span class="ot">=</span> <span class="st">"#31A354"</span>,</span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_1_5"</span> <span class="ot">=</span> <span class="st">"#74C476"</span>, <span class="st">"default_1"</span> <span class="ot">=</span> <span class="st">"#BAE4B3"</span>)</span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>name, <span class="at">ncol =</span> <span class="dv">4</span>, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"OrthoFinder mode"</span>,</span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Relationship between the number of orthogroups and orthogroup size per OrthoFinder mode"</span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>percentage_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>It is very clear that increasing the mcl inflation increases the number of orthogroups, but decreases the percentage of OGs with more than 100 and 50 genes.</p>
<p>Finally, let’s remove OGs with &gt;=200 genes to remove noise.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter OGs</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>ogs_filtered <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(ogs), <span class="cf">function</span>(x) {</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Which OGs less than 200 genes?</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    og_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(og_sizes[[x]][og_sizes[[x]]<span class="sc">$</span>Total <span class="sc">&lt;</span> <span class="dv">200</span>, ])</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    fogs <span class="ot">&lt;-</span> ogs[[x]][ogs[[x]]<span class="sc">$</span>Orthogroup <span class="sc">%in%</span> og_keep, ]</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(fogs)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ogs_filtered) <span class="ot">&lt;-</span> <span class="fu">names</span>(ogs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="orthogroup-assessment" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="orthogroup-assessment"><span class="header-section-number">4.3</span> Orthogroup assessment</h2>
<p>Now, let’s get InterPro domain annotation for the following species to assess orthogroups:</p>
<ul>
<li><em>A. thaliana</em></li>
<li><em>A. arabicum</em></li>
<li><em>A. lyrata</em></li>
<li><em>B. carinata</em></li>
<li><em>C. rubella</em></li>
<li><em>C. hirsuta</em></li>
<li><em>S. parvula</em></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define function to read functional annotation from PLAZA 5.0</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>read_annotation <span class="ot">&lt;-</span> <span class="cf">function</span>(url, <span class="at">cols =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)) {</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    annot <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(url, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">skip =</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(cols)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(annot)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Gene"</span>, <span class="st">"Annotation"</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(annot)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Interpro annotation</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>base <span class="ot">&lt;-</span> <span class="st">"https://ftp.psb.ugent.be/pub/plaza/plaza_public_dicots_05/InterPro/"</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>interpro <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Athaliana =</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(base, <span class="st">"interpro.ath.csv.gz"</span>)),     </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Aarabicum =</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(base, <span class="st">"interpro.aar.csv.gz"</span>)),</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">Alyrata_cvMN47 =</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(base, <span class="st">"interpro.aly.csv.gz"</span>)),</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">Bcarinata_cvzd1 =</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(base, <span class="st">"interpro.bca.csv.gz"</span>)),</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">Crubella_cvMonteGargano =</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(base, <span class="st">"interpro.cru.csv.gz"</span>)),</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">Chirsuta =</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(base, <span class="st">"interpro.chi.csv.gz"</span>)),</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sparvula =</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(base, <span class="st">"interpro.spa.csv.gz"</span>))</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>interpro <span class="ot">&lt;-</span> <span class="fu">lapply</span>(interpro, as.data.frame)</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate homogeneity scores</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>species_annotation <span class="ot">&lt;-</span> <span class="fu">names</span>(interpro)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>og_assessment <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(ogs_filtered), <span class="cf">function</span>(x) {</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Working on mode "</span>, <span class="fu">names</span>(ogs_filtered)[x])</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>    orthogroups <span class="ot">&lt;-</span> ogs_filtered[[x]]</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>    orthogroups <span class="ot">&lt;-</span> orthogroups[orthogroups<span class="sc">$</span>Species <span class="sc">%in%</span> species_annotation, ]</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">assess_orthogroups</span>(orthogroups, interpro)</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>    res<span class="sc">$</span>Mode <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(ogs_filtered)[x], </span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ultra_3"</span>, <span class="st">"ultra_2"</span>, <span class="st">"ultra_1_5"</span>, <span class="st">"ultra_1"</span>,</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>            <span class="st">"default_3"</span>, <span class="st">"default_2"</span>, <span class="st">"default_1_5"</span>, <span class="st">"default_1"</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>og_assessment <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, og_assessment)</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Save homogeneity stats</span></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>    og_assessment, <span class="at">compress =</span> <span class="st">"xz"</span>,</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"result_files"</span>, <span class="st">"og_assessment_brassicaceae.rda"</span>)</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="comparing-and-visualizing-homogeneity-statistics" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="comparing-and-visualizing-homogeneity-statistics"><span class="header-section-number">4.4</span> Comparing and visualizing homogeneity statistics</h2>
<p>Here, we will compare and visualize how the homogeneity scores are affected by:</p>
<ul>
<li>different species choice</li>
<li>different mcl inflation values</li>
<li>different DIAMOND modes (default and ultra)</li>
</ul>
<p>Quick exploration of median and mean homogeneity:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"result_files"</span>, <span class="st">"og_assessment_brassicaceae.rda"</span>))</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale value to the maximum so that values range from 0 to 1</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>og_assessment<span class="sc">$</span>Median_score <span class="ot">&lt;-</span> og_assessment<span class="sc">$</span>Median_score <span class="sc">/</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">max</span>(og_assessment<span class="sc">$</span>Median_score)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>mean_og <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Mode) <span class="sc">%&gt;%</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(Median_score))</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Median</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>median_og <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Mode) <span class="sc">%&gt;%</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">median =</span> <span class="fu">median</span>(Median_score))</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>mean_and_median_og <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(mean_og, median_og) <span class="sc">|&gt;</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Mean =</span> mean, <span class="at">Median =</span> median)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(mean_and_median_og, <span class="at">caption =</span> <span class="st">"Mean and median OG scores."</span>, <span class="at">digits =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Mean and median OG scores.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Mode</th>
<th style="text-align: right;">Mean</th>
<th style="text-align: right;">Median</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: right;">0.640</td>
<td style="text-align: right;">0.640</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_2</td>
<td style="text-align: right;">0.631</td>
<td style="text-align: right;">0.635</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: right;">0.620</td>
<td style="text-align: right;">0.628</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_1</td>
<td style="text-align: right;">0.425</td>
<td style="text-align: right;">0.424</td>
</tr>
<tr class="odd">
<td style="text-align: left;">default_3</td>
<td style="text-align: right;">0.639</td>
<td style="text-align: right;">0.640</td>
</tr>
<tr class="even">
<td style="text-align: left;">default_2</td>
<td style="text-align: right;">0.631</td>
<td style="text-align: right;">0.635</td>
</tr>
<tr class="odd">
<td style="text-align: left;">default_1_5</td>
<td style="text-align: right;">0.620</td>
<td style="text-align: right;">0.628</td>
</tr>
<tr class="even">
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">0.425</td>
<td style="text-align: right;">0.423</td>
</tr>
</tbody>
</table>
</div>
</div>
<section id="global-distributions" class="level3" data-number="4.4.1">
<h3 data-number="4.4.1" class="anchored" data-anchor-id="global-distributions"><span class="header-section-number">4.4.1</span> Global distributions</h3>
<p>Here, we will compare and visualize all distros considering different DIAMOND modes and mcl inflation values. To start, let’s perform Wilcoxon tests for all combinations of modes and obtain effect sizes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Relevel 'Mode' factor</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>og_assessment<span class="sc">$</span>Mode <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    og_assessment<span class="sc">$</span>Mode, </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ultra_3"</span>, <span class="st">"ultra_2"</span>, <span class="st">"ultra_1_5"</span>, <span class="st">"ultra_1"</span>,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"default_3"</span>, <span class="st">"default_2"</span>, <span class="st">"default_1_5"</span>, <span class="st">"default_1"</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Comparing all vs all</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>comp_global <span class="ot">&lt;-</span> <span class="fu">compare</span>(og_assessment, <span class="st">"Median_score ~ Mode"</span>)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>comp_global <span class="sc">|&gt;</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter_comparison</span>() <span class="sc">|&gt;</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">"Mann-Whitney U test for differences in orthogroup scores with Wilcoxon effect sizes."</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">10</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Mann-Whitney U test for differences in orthogroup scores with Wilcoxon effect sizes.</caption>
<colgroup>
<col style="width: 18%">
<col style="width: 18%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 13%">
<col style="width: 16%">
<col style="width: 15%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">group1</th>
<th style="text-align: left;">group2</th>
<th style="text-align: right;">n1</th>
<th style="text-align: right;">n2</th>
<th style="text-align: right;">padj</th>
<th style="text-align: right;">effsize</th>
<th style="text-align: left;">magnitude</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: left;">ultra_2</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04120347</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.06125964</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: left;">ultra_1</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.34087185</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: left;">default_3</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">5.00e-10</td>
<td style="text-align: right;">0.03113134</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: left;">default_2</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04197169</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: left;">default_1_5</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.06233198</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_3</td>
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.34258513</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_2</td>
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04340346</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_2</td>
<td style="text-align: left;">ultra_1</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.33536590</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_2</td>
<td style="text-align: left;">default_3</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04018176</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_2</td>
<td style="text-align: left;">default_2</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">2.55e-08</td>
<td style="text-align: right;">0.02855053</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_2</td>
<td style="text-align: left;">default_1_5</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04451653</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_2</td>
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.33742024</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: left;">ultra_1</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.32401575</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: left;">default_3</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.05737302</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: left;">default_2</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04259062</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: left;">default_1_5</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">1.72e-06</td>
<td style="text-align: right;">0.02554732</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_1_5</td>
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.32541386</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_1</td>
<td style="text-align: left;">default_3</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.34014551</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_1</td>
<td style="text-align: left;">default_2</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.33456485</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ultra_1</td>
<td style="text-align: left;">default_1_5</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.32260079</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra_1</td>
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">2.70e-02</td>
<td style="text-align: right;">0.01928759</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">default_3</td>
<td style="text-align: left;">default_2</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04084430</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">default_3</td>
<td style="text-align: left;">default_1_5</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.06124558</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">default_3</td>
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.34147823</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">default_2</td>
<td style="text-align: left;">default_1_5</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.04366051</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">default_2</td>
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.33616829</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">default_1_5</td>
<td style="text-align: left;">default_1</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.32420710</td>
<td style="text-align: left;">moderate</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As we can see, using mcl = 1 leads to much smaller homogeneity scores as compared to every other mcl value. For mcl values &gt;=1.5, there are differences, but they are likely due to large sample sizes, as indicated by small effect sizes.</p>
<p>The default OrthoFinder mode (default DIAMOND, mcl = 1.5) leads to higher homogeneity as compared to runs using mcl = 1, both in default and ultrasensitive DIAMOND modes. The difference between the default mode and runs with higher mcl values are negligible.</p>
<p>Now, let’s visualize the distributions and compare the default OrthoFinder mode with every other mode, highlighting significant differences (<em>P</em> &lt; 0.05) with effect size &gt; 0.1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>global_comps <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"default_1_5"</span>, <span class="st">"ultra_1"</span>),</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"default_1_5"</span>, <span class="st">"default_1"</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>p_distros_global <span class="ot">&lt;-</span> <span class="fu">ggviolin</span>(</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    og_assessment, <span class="at">y =</span> <span class="st">"Median_score"</span>, <span class="at">x =</span> <span class="st">"Mode"</span>, </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">orientation =</span> <span class="st">"horiz"</span>, <span class="at">trim =</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">add =</span> <span class="fu">c</span>(<span class="st">"boxplot"</span>, <span class="st">"mean"</span>), </span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Mode"</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">fill =</span> <span class="st">"white"</span>)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"ultra_3"</span> <span class="ot">=</span> <span class="st">"#08519C"</span>, <span class="st">"ultra_2"</span> <span class="ot">=</span> <span class="st">"#3182BD"</span>,</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"ultra_1_5"</span> <span class="ot">=</span> <span class="st">"#6BAED6"</span>, <span class="st">"ultra_1"</span> <span class="ot">=</span> <span class="st">"#BDD7E7"</span>,</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_3"</span> <span class="ot">=</span> <span class="st">"#006D2C"</span>, <span class="st">"default_2"</span> <span class="ot">=</span> <span class="st">"#31A354"</span>,</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_1_5"</span> <span class="ot">=</span> <span class="st">"#74C476"</span>, <span class="st">"default_1"</span> <span class="ot">=</span> <span class="st">"#BAE4B3"</span>)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>(</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">comparisons =</span> global_comps, <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"wilcox.test"</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Scaled homogeneity scores"</span>, <span class="at">x =</span> <span class="st">"OrthoFinder modes"</span>,</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Distribution of mean homogeneity scores for orthogroups"</span>) <span class="sc">+</span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.subtitle =</span> ggtext<span class="sc">::</span><span class="fu">element_markdown</span>())</span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>p_distros_global</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/og-distros-viz-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Distribution of mean orthogroup scores for each OrthoFinder run.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="the-effect-of-species-choice" class="level3" data-number="4.4.2">
<h3 data-number="4.4.2" class="anchored" data-anchor-id="the-effect-of-species-choice"><span class="header-section-number">4.4.2</span> The effect of species choice</h3>
<p>Here, we will compare the distributions of orthogroups scores using each species individually to see if the species choice has an impact on the conclusions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>og_species_long <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">8</span>, <span class="cf">function</span>(x) {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    var <span class="ot">&lt;-</span> <span class="fu">names</span>(og_assessment)[x]</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    species_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, var)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>    long_df <span class="ot">&lt;-</span> og_assessment[, <span class="fu">c</span>(<span class="st">"Orthogroups"</span>, var, <span class="st">"Mode"</span>)]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(long_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"OGs"</span>, <span class="st">"Score"</span>, <span class="st">"Mode"</span>)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>    long_df<span class="sc">$</span>Score <span class="ot">&lt;-</span> long_df<span class="sc">$</span>Score <span class="sc">/</span> <span class="fu">max</span>(long_df<span class="sc">$</span>Score, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    long_df<span class="sc">$</span>Species <span class="ot">&lt;-</span> species_name</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(long_df)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>og_species_long <span class="ot">&lt;-</span> og_species_long[<span class="sc">!</span><span class="fu">is.na</span>(og_species_long<span class="sc">$</span>Score), ]</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>og_species_long <span class="ot">&lt;-</span> og_species_long <span class="sc">|&gt;</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">Species =</span> <span class="fu">str_replace_all</span>(</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>            Species, </span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Aarabicum"</span> <span class="ot">=</span> <span class="st">"A. arabicum"</span>,</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Alyrata"</span> <span class="ot">=</span> <span class="st">"A. lyrata"</span>,</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Athaliana"</span> <span class="ot">=</span> <span class="st">"A. thaliana"</span>,</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Bcarinata"</span> <span class="ot">=</span> <span class="st">"B. carinata"</span>,</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Chirsuta"</span> <span class="ot">=</span> <span class="st">"C. hirsuta"</span>,</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Crubella"</span> <span class="ot">=</span> <span class="st">"C. rubella"</span>,</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Sparvula"</span> <span class="ot">=</span> <span class="st">"S. parvula"</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>p_distros_by_species <span class="ot">&lt;-</span> <span class="fu">ggviolin</span>(</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>    og_species_long, </span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Score"</span>, <span class="at">x =</span> <span class="st">"Mode"</span>, </span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">orientation =</span> <span class="st">"horiz"</span>, <span class="at">trim =</span> <span class="cn">TRUE</span>,</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">add =</span> <span class="fu">c</span>(<span class="st">"boxplot"</span>, <span class="st">"mean"</span>), <span class="at">facet.by =</span> <span class="st">"Species"</span>, <span class="at">nrow =</span> <span class="dv">1</span>,</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Mode"</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">fill =</span> <span class="st">"white"</span>)</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ultra_3"</span> <span class="ot">=</span> <span class="st">"#08519C"</span>, <span class="st">"ultra_2"</span> <span class="ot">=</span> <span class="st">"#3182BD"</span>,</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ultra_1_5"</span> <span class="ot">=</span> <span class="st">"#6BAED6"</span>, <span class="st">"ultra_1"</span> <span class="ot">=</span> <span class="st">"#BDD7E7"</span>,</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>            <span class="st">"default_3"</span> <span class="ot">=</span> <span class="st">"#006D2C"</span>, <span class="st">"default_2"</span> <span class="ot">=</span> <span class="st">"#31A354"</span>,</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>            <span class="st">"default_1_5"</span> <span class="ot">=</span> <span class="st">"#74C476"</span>, <span class="st">"default_1"</span> <span class="ot">=</span> <span class="st">"#BAE4B3"</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Scaled homogeneity scores"</span>, <span class="at">x =</span> <span class="st">"OrthoFinder modes"</span>,</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Distribution of OG scores for each species"</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(</span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>            <span class="st">"default_1"</span> <span class="ot">=</span> <span class="st">"Default, 1"</span>,</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>            <span class="st">"default_1_5"</span> <span class="ot">=</span> <span class="st">"Default, 1.5"</span>,</span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>            <span class="st">"default_2"</span> <span class="ot">=</span> <span class="st">"Default, 2"</span>,</span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>            <span class="st">"default_3"</span> <span class="ot">=</span> <span class="st">"Default, 3"</span>,</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ultra_1"</span> <span class="ot">=</span> <span class="st">"Ultra, 1"</span>,</span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ultra_1_5"</span> <span class="ot">=</span> <span class="st">"Ultra, 1.5"</span>,</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ultra_2"</span> <span class="ot">=</span> <span class="st">"Ultra, 2"</span>,</span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ultra_3"</span> <span class="ot">=</span> <span class="st">"Ultra, 3"</span></span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>p_distros_by_species</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/species-choice-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Distribution of orthogroup scores for each OrthoFinder run calculated for each species separately.</figcaption>
</figure>
</div>
</div>
</div>
<p>We conclude that the species choice does not affect the comparisons of orthogroup scores among OrthoFinder runs.</p>
</section>
<section id="the-effect-of-mcl-inflation-parameters" class="level3" data-number="4.4.3">
<h3 data-number="4.4.3" class="anchored" data-anchor-id="the-effect-of-mcl-inflation-parameters"><span class="header-section-number">4.4.3</span> The effect of mcl inflation parameters</h3>
<p>Here, we will explore the impact of changing mcl inflation parameters in the homogeneity of orthogroups.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Process data to include information on DIAMOND mode and mcl</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>og_modes <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">diamond =</span> <span class="fu">str_replace_all</span>(Mode, <span class="st">"_.*"</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="fu">str_replace_all</span>(Mode, <span class="fu">c</span>(<span class="st">"default_"</span> <span class="ot">=</span> <span class="st">""</span>, <span class="st">"ultra_"</span> <span class="ot">=</span> <span class="st">""</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="fu">str_replace_all</span>(mcl, <span class="st">"_"</span>, <span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="fu">as.numeric</span>(mcl))</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtain P-values from Wilcoxon tests and effect sizes</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>comp_mcl_default <span class="ot">&lt;-</span> og_modes <span class="sc">%&gt;%</span> </span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(diamond <span class="sc">==</span> <span class="st">"default"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">compare</span>(., <span class="st">"Median_score ~ mcl"</span>)</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>comp_mcl_default <span class="sc">|&gt;</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter_comparison</span>() <span class="sc">|&gt;</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">"Mann-Whitney U test for differences in orthogroup scores between runs with different mcl parameters and standard DIAMOND mode. Effect sizes represent Wilcoxon effect sizes."</span>,</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">10</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Mann-Whitney U test for differences in orthogroup scores between runs with different mcl parameters and standard DIAMOND mode. Effect sizes represent Wilcoxon effect sizes.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">group1</th>
<th style="text-align: left;">group2</th>
<th style="text-align: right;">n1</th>
<th style="text-align: right;">n2</th>
<th style="text-align: right;">padj</th>
<th style="text-align: right;">effsize</th>
<th style="text-align: left;">magnitude</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">1.5</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.32420710</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">1</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.33616829</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.34147823</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">1.5</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.04366051</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1.5</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.06124558</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.04084430</td>
<td style="text-align: left;">small</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>comp_mcl_ultra <span class="ot">&lt;-</span> og_modes <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(diamond <span class="sc">==</span> <span class="st">"ultra"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">compare</span>(., <span class="st">"Median_score ~ mcl"</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>comp_mcl_ultra <span class="sc">|&gt;</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter_comparison</span>() <span class="sc">|&gt;</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">"Mann-Whitney U test for differences in orthogroup scores between runs with different mcl parameters and ultra-sensitive DIAMOND mode. Effect sizes represent Wilcoxon effect sizes."</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">10</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Mann-Whitney U test for differences in orthogroup scores between runs with different mcl parameters and ultra-sensitive DIAMOND mode. Effect sizes represent Wilcoxon effect sizes.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">group1</th>
<th style="text-align: left;">group2</th>
<th style="text-align: right;">n1</th>
<th style="text-align: right;">n2</th>
<th style="text-align: right;">padj</th>
<th style="text-align: right;">effsize</th>
<th style="text-align: left;">magnitude</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">1.5</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.32401575</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">1</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.33536590</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.34087185</td>
<td style="text-align: left;">moderate</td>
</tr>
<tr class="even">
<td style="text-align: left;">1.5</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.04340346</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1.5</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.06125964</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.04120347</td>
<td style="text-align: left;">small</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In line with what we demonstrated in the global distributions, the Wilcoxon tests show that mcl = 1 leads to much lower homogeneity scores than all other mcl values, regardless of the DIAMOND mode. Additionally, increasing mcl values leads to increased homogeneity scores (i.e., homogeneity scores follow the order of mcl 3 &gt; 2 &gt; 1.5 &gt; 1), but differences among mcl values &gt;=1.5 are negligible, as indicated by small effect sizes. Thus, low P-values could be due to large sample sizes.</p>
<p>Now, let’s visualize the distributions.</p>
<div class="cell" data-fig.dpi="200">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List of comparisons to be made</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>mcl_comp <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"1.5"</span>), <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>), <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"3"</span>), <span class="fu">c</span>(<span class="st">"1.5"</span>, <span class="st">"3"</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>p_distros_mcl <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">diamond =</span> <span class="fu">str_replace_all</span>(Mode, <span class="st">"_.*"</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="fu">str_replace_all</span>(Mode, <span class="fu">c</span>(<span class="st">"default_"</span> <span class="ot">=</span> <span class="st">""</span>, <span class="st">"ultra_"</span> <span class="ot">=</span> <span class="st">""</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="fu">str_replace_all</span>(mcl, <span class="st">"_"</span>, <span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="fu">as.numeric</span>(mcl)) <span class="sc">%&gt;%</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggviolin</span>(., <span class="at">x =</span> <span class="st">"mcl"</span>, <span class="at">y =</span> <span class="st">"Median_score"</span>, <span class="at">trim =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">add =</span> <span class="fu">c</span>(<span class="st">"boxplot"</span>, <span class="st">"mean"</span>), <span class="at">facet.by =</span> <span class="st">"diamond"</span>,</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill =</span> <span class="st">"Mode"</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">fill =</span> <span class="st">"white"</span>)) <span class="sc">+</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"ultra_3"</span> <span class="ot">=</span> <span class="st">"#08519C"</span>, <span class="st">"ultra_2"</span> <span class="ot">=</span> <span class="st">"#3182BD"</span>,</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"ultra_1_5"</span> <span class="ot">=</span> <span class="st">"#6BAED6"</span>, <span class="st">"ultra_1"</span> <span class="ot">=</span> <span class="st">"#BDD7E7"</span>,</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_3"</span> <span class="ot">=</span> <span class="st">"#006D2C"</span>, <span class="st">"default_2"</span> <span class="ot">=</span> <span class="st">"#31A354"</span>,</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_1_5"</span> <span class="ot">=</span> <span class="st">"#74C476"</span>, <span class="st">"default_1"</span> <span class="ot">=</span> <span class="st">"#BAE4B3"</span>)</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>(</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">comparisons =</span> mcl_comp, <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"wilcox.test"</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Scaled homogeneity scores"</span>, <span class="at">x =</span> <span class="st">"MCL inflation parameters"</span>,</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Effect of MCL inflation values on orthogroup inference"</span>,</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"Panels represent DIAMOND sensitivity modes"</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>p_distros_mcl</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/viz-og-mcl-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Effect of MCL inflation values on orthogroup scores.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="the-effect-of-diamond-mode-default-vs-ultra" class="level3" data-number="4.4.4">
<h3 data-number="4.4.4" class="anchored" data-anchor-id="the-effect-of-diamond-mode-default-vs-ultra"><span class="header-section-number">4.4.4</span> The effect of DIAMOND mode (default vs ultra)</h3>
<p>Here, we will investigate whether changing the DIAMOND mode (default vs ultrasensitive) in OrthoFinder affects orthogroup homogeneity.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare median scores</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>mcl1 <span class="ot">&lt;-</span> og_modes <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(mcl <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">compare</span>(., <span class="st">"Median_score ~ diamond"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter_comparison</span>()</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>mcl1_5 <span class="ot">&lt;-</span> og_modes <span class="sc">%&gt;%</span> </span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(mcl <span class="sc">==</span> <span class="fl">1.5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">compare</span>(., <span class="st">"Median_score ~ diamond"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter_comparison</span>()</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>mcl2 <span class="ot">&lt;-</span> og_modes <span class="sc">%&gt;%</span> </span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(mcl <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">compare</span>(., <span class="st">"Median_score ~ diamond"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter_comparison</span>()</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>mcl3 <span class="ot">&lt;-</span> og_modes <span class="sc">%&gt;%</span> </span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(mcl <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">compare</span>(., <span class="st">"Median_score ~ diamond"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter_comparison</span>()</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>    mcl1 <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="dv">1</span>),</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>    mcl1_5 <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="fl">1.5</span>), </span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>    mcl2 <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="dv">2</span>), </span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>    mcl3 <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">mcl =</span> <span class="dv">3</span>)</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">"Mann-Whitney U test for differences in orthogroup scores between runs with different DIAMOND modes for each mcl value. Effect sizes represent Wilcoxon effect sizes."</span>,</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">10</span></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Mann-Whitney U test for differences in orthogroup scores between runs with different DIAMOND modes for each mcl value. Effect sizes represent Wilcoxon effect sizes.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">group1</th>
<th style="text-align: left;">group2</th>
<th style="text-align: right;">n1</th>
<th style="text-align: right;">n2</th>
<th style="text-align: right;">padj</th>
<th style="text-align: right;">effsize</th>
<th style="text-align: left;">magnitude</th>
<th style="text-align: right;">mcl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">default</td>
<td style="text-align: left;">ultra</td>
<td style="text-align: right;">5587</td>
<td style="text-align: right;">5534</td>
<td style="text-align: right;">2.10e-02</td>
<td style="text-align: right;">0.01928759</td>
<td style="text-align: left;">small</td>
<td style="text-align: right;">1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">default</td>
<td style="text-align: left;">ultra</td>
<td style="text-align: right;">16975</td>
<td style="text-align: right;">16898</td>
<td style="text-align: right;">1.29e-06</td>
<td style="text-align: right;">0.02554732</td>
<td style="text-align: left;">small</td>
<td style="text-align: right;">1.5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">default</td>
<td style="text-align: left;">ultra</td>
<td style="text-align: right;">18633</td>
<td style="text-align: right;">18575</td>
<td style="text-align: right;">1.82e-08</td>
<td style="text-align: right;">0.02855053</td>
<td style="text-align: left;">small</td>
<td style="text-align: right;">2.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">default</td>
<td style="text-align: left;">ultra</td>
<td style="text-align: right;">19765</td>
<td style="text-align: right;">19738</td>
<td style="text-align: right;">3.00e-10</td>
<td style="text-align: right;">0.03113134</td>
<td style="text-align: left;">small</td>
<td style="text-align: right;">3.0</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Again, we can see that there are significant P-values, but very small effect sizes, indicating no difference resulting from the DIAMOND mode. Thus, users can run the default mode of DIAMOND, which is way faster, without any loss of biological signal for orthogroup inference.</p>
<p>Let’s visualize the distributions.</p>
<div class="cell" data-fig.dpi="200">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>p_distros_diamond <span class="ot">&lt;-</span> og_modes <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggviolin</span>(., <span class="at">x =</span> <span class="st">"diamond"</span>, <span class="at">y =</span> <span class="st">"Median_score"</span>, <span class="at">trim =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">add =</span> <span class="fu">c</span>(<span class="st">"boxplot"</span>, <span class="st">"mean"</span>), <span class="at">facet.by =</span> <span class="st">"mcl"</span>, <span class="at">ncol =</span> <span class="dv">4</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill =</span> <span class="st">"Mode"</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">fill =</span> <span class="st">"white"</span>)) <span class="sc">+</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"ultra_3"</span> <span class="ot">=</span> <span class="st">"#08519C"</span>, <span class="st">"ultra_2"</span> <span class="ot">=</span> <span class="st">"#3182BD"</span>,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"ultra_1_5"</span> <span class="ot">=</span> <span class="st">"#6BAED6"</span>, <span class="st">"ultra_1"</span> <span class="ot">=</span> <span class="st">"#BDD7E7"</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_3"</span> <span class="ot">=</span> <span class="st">"#006D2C"</span>, <span class="st">"default_2"</span> <span class="ot">=</span> <span class="st">"#31A354"</span>,</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"default_1_5"</span> <span class="ot">=</span> <span class="st">"#74C476"</span>, <span class="st">"default_1"</span> <span class="ot">=</span> <span class="st">"#BAE4B3"</span>)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Scaled homogeneity scores"</span>, <span class="at">x =</span> <span class="st">"DIAMOND mode"</span>,</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Effect of DIAMOND sensitivity mode on orthogroup inference"</span>,</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">subtitle =</span> <span class="st">"Panels represent MCL inflation parameters"</span>) <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.subtitle =</span> ggtext<span class="sc">::</span><span class="fu">element_markdown</span>())</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>p_distros_diamond</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/og-diamond-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Effect of DIAMOND mode on orthogroup scores.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="functional-analysis-of-homogeneous-and-heterogeneous-gene-families" class="level2" data-number="4.5">
<h2 data-number="4.5" class="anchored" data-anchor-id="functional-analysis-of-homogeneous-and-heterogeneous-gene-families"><span class="header-section-number">4.5</span> Functional analysis of homogeneous and heterogeneous gene families</h2>
<p>By looking at the global distributions of homogeneity scores, we can see that all distributions have a similar shape. This pattern suggests that some gene families tend to be more homogeneous (scores close to 1), while others tend to include domains that are not shared by all members. The latter can be, for instance, rapidly evolving families that gain or lose domains at faster rates.</p>
<p>To explore what these groups of families contain, we will perform a functional enrichment analysis each group. First of anything, let’s plot the distribution for the default OrthoFinder mode and highlight the groups.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot distro with groups</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>p_distros_groups <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Mode <span class="sc">==</span> <span class="st">"default_1_5"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Median_score)) <span class="sc">+</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>(<span class="at">fill =</span> <span class="st">"grey80"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>    ggpubr<span class="sc">::</span><span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Density"</span>, <span class="at">x =</span> <span class="st">"Orthogroup scores"</span>,</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Distribution of mean homogeneity scores for orthogroups"</span>,</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"Scores for the default OrthoFinder mode"</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.56</span>, <span class="at">color =</span> <span class="st">"firebrick"</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.87</span>, <span class="at">color =</span> <span class="st">"firebrick"</span>, <span class="at">linetype =</span> <span class="dv">2</span>)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>p_distros_groups</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/plot-distros-with-groups-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Distribution of mean homogeneity scores for orthogroups</figcaption>
</figure>
</div>
</div>
</div>
<p>Now, let’s get vectors of genes in orthogroups from each of the groups highlighted in the figure above.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>species <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Athaliana"</span>, <span class="st">"Aarabicum"</span>, <span class="st">"Alyrata_cvMN47"</span>, <span class="st">"Bcarinata_cvzd1"</span>,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Crubella_cvMonteGargano"</span>, <span class="st">"Chirsuta"</span>, <span class="st">"Sparvula"</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get genes and orthogroups (default mode)</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>genes_ogs <span class="ot">&lt;-</span> ogs_filtered<span class="sc">$</span>default_1_5</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only species for which we have functional annotation info</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>genes_ogs <span class="ot">&lt;-</span> genes_ogs[genes_ogs<span class="sc">$</span>Species <span class="sc">%in%</span> species, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)]</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Get background genes (all genes in OGs)</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>background <span class="ot">&lt;-</span> genes_ogs<span class="sc">$</span>Gene</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Find orthogroups for each group</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="do">## G1: 0 - 0.56</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Mode <span class="sc">==</span> <span class="st">"default_1_5"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Median_score =</span> Median_score <span class="sc">/</span> <span class="fu">max</span>(Median_score)) <span class="sc">%&gt;%</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Median_score <span class="sc">&lt;=</span> <span class="fl">0.56</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(Orthogroups) <span class="sc">%&gt;%</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(., genes_ogs, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Orthogroups"</span> <span class="ot">=</span> <span class="st">"Orthogroup"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Gene)</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="do">## G2: 0.56 - 0.87</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Mode <span class="sc">==</span> <span class="st">"default_1_5"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Median_score =</span> Median_score <span class="sc">/</span> <span class="fu">max</span>(Median_score)) <span class="sc">%&gt;%</span></span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Median_score <span class="sc">&gt;</span> <span class="fl">0.56</span> <span class="sc">&amp;</span> Median_score <span class="sc">&lt;=</span> <span class="fl">0.87</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(Orthogroups) <span class="sc">%&gt;%</span></span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(., genes_ogs, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Orthogroups"</span> <span class="ot">=</span> <span class="st">"Orthogroup"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Gene)</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a><span class="do">## G3: 0.87 - 1</span></span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>g3 <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Mode <span class="sc">==</span> <span class="st">"default_1_5"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Median_score =</span> Median_score <span class="sc">/</span> <span class="fu">max</span>(Median_score)) <span class="sc">%&gt;%</span></span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Median_score <span class="sc">&gt;</span> <span class="fl">0.87</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(Orthogroups) <span class="sc">%&gt;%</span></span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(., genes_ogs, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Orthogroups"</span> <span class="ot">=</span> <span class="st">"Orthogroup"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we need to get functional annotation from PLAZA.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">timeout =</span> <span class="dv">6000</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>plaza_species <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ath"</span>, <span class="st">"aar"</span>, <span class="st">"aly"</span>, <span class="st">"bca"</span>, <span class="st">"cru"</span>, <span class="st">"chi"</span>, <span class="st">"spa"</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># GO annotation</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>bgo <span class="ot">&lt;-</span> <span class="st">"https://ftp.psb.ugent.be/pub/plaza/plaza_public_dicots_05/GO/"</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>go <span class="ot">&lt;-</span> <span class="fu">lapply</span>(plaza_species, <span class="cf">function</span>(x) {</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(bgo, <span class="st">"go."</span>, x, <span class="st">".csv.gz"</span>), <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">8</span>))</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>    term2gene <span class="ot">&lt;-</span> y[, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>)] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(., <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>    term2name <span class="ot">&lt;-</span> y[, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>)] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(., <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2GENE =</span> <span class="fu">as.data.frame</span>(term2gene), </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2NAME =</span> <span class="fu">as.data.frame</span>(term2name)</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>go_gene <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(go, <span class="cf">function</span>(x) <span class="fu">return</span>(x<span class="sc">$</span>TERM2GENE)))</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>go_des <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(go, <span class="cf">function</span>(x) <span class="fu">return</span>(x<span class="sc">$</span>TERM2NAME)))</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Remove non-BP terms</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>ath_bp <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">"ath_bp.rds"</span>)</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"https://jokergoo.github.io/rGREAT_genesets/genesets/bp_athaliana_eg_gene_go_genesets.rds"</span>,</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">destfile =</span> ath_bp</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>gobp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(ath_bp)</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>gobp <span class="ot">&lt;-</span> <span class="fu">names</span>(gobp)</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>go_gene <span class="ot">&lt;-</span> go_gene[go_gene<span class="sc">$</span>Annotation <span class="sc">%in%</span> gobp, ]</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>go_des <span class="ot">&lt;-</span> go_des[go_des<span class="sc">$</span>Annotation <span class="sc">%in%</span> gobp, ]</span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(gobp)</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a><span class="co"># MapMan annotation</span></span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>bmm <span class="ot">&lt;-</span> <span class="st">"https://ftp.psb.ugent.be/pub/plaza/plaza_public_dicots_05/MapMan/"</span></span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>mm <span class="ot">&lt;-</span> <span class="fu">lapply</span>(plaza_species, <span class="cf">function</span>(x) {</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(bmm, <span class="st">"mapman."</span>, x, <span class="st">".csv.gz"</span>), <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>))</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>    term2gene <span class="ot">&lt;-</span> y[, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>)] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(., <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>    term2name <span class="ot">&lt;-</span> y[, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>)] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(., <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2GENE =</span> <span class="fu">as.data.frame</span>(term2gene), </span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2NAME =</span> <span class="fu">as.data.frame</span>(term2name)</span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a>mm_gene <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(mm, <span class="cf">function</span>(x) <span class="fu">return</span>(x<span class="sc">$</span>TERM2GENE)))</span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a>mm_des <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(mm, <span class="cf">function</span>(x) <span class="fu">return</span>(x<span class="sc">$</span>TERM2NAME))) <span class="sc">%&gt;%</span></span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">desc =</span> <span class="fu">str_replace_all</span>(desc, <span class="st">".*</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">""</span>))</span>
<span id="cb21-46"><a href="#cb21-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-47"><a href="#cb21-47" aria-hidden="true" tabindex="-1"></a><span class="co"># InterPro</span></span>
<span id="cb21-48"><a href="#cb21-48" aria-hidden="true" tabindex="-1"></a>bi <span class="ot">&lt;-</span> <span class="st">"https://ftp.psb.ugent.be/pub/plaza/plaza_public_dicots_05/InterPro/"</span></span>
<span id="cb21-49"><a href="#cb21-49" aria-hidden="true" tabindex="-1"></a>ip <span class="ot">&lt;-</span> <span class="fu">lapply</span>(plaza_species, <span class="cf">function</span>(x) {</span>
<span id="cb21-50"><a href="#cb21-50" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> <span class="fu">read_annotation</span>(<span class="fu">paste0</span>(bi, <span class="st">"interpro."</span>, x, <span class="st">".csv.gz"</span>), <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">4</span>))</span>
<span id="cb21-51"><a href="#cb21-51" aria-hidden="true" tabindex="-1"></a>    term2gene <span class="ot">&lt;-</span> y[, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>)] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(., <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-52"><a href="#cb21-52" aria-hidden="true" tabindex="-1"></a>    term2name <span class="ot">&lt;-</span> y[, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>)] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(., <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-53"><a href="#cb21-53" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb21-54"><a href="#cb21-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2GENE =</span> <span class="fu">as.data.frame</span>(term2gene), </span>
<span id="cb21-55"><a href="#cb21-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2NAME =</span> <span class="fu">as.data.frame</span>(term2name)</span>
<span id="cb21-56"><a href="#cb21-56" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb21-57"><a href="#cb21-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb21-58"><a href="#cb21-58" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb21-59"><a href="#cb21-59" aria-hidden="true" tabindex="-1"></a>ip_gene <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(ip, <span class="cf">function</span>(x) <span class="fu">return</span>(x<span class="sc">$</span>TERM2GENE)))</span>
<span id="cb21-60"><a href="#cb21-60" aria-hidden="true" tabindex="-1"></a>ip_des <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(ip, <span class="cf">function</span>(x) <span class="fu">return</span>(x<span class="sc">$</span>TERM2NAME)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we can finally perform the enrichment analyses.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform enrichment analyses</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>tgene <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">GO =</span> go_gene, </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">MapMan =</span> mm_gene,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">InterPro =</span> ip_gene</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>tname <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">GO =</span> go_des,</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">MapMan =</span> mm_des,</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">InterPro =</span> ip_des</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="do">## G1</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>g1_sea <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(<span class="fu">seq_along</span>(tgene), <span class="cf">function</span>(x) {</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="fu">enricher</span>(</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>        g1, <span class="at">universe =</span> background, </span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2GENE =</span> tgene[[x]], <span class="at">TERM2NAME =</span> tname[[x]]</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>    ))[, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="do">## G2</span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>g2_sea <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(<span class="fu">seq_along</span>(tgene), <span class="cf">function</span>(x) {</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="fu">enricher</span>(</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>        g2, <span class="at">universe =</span> background, </span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2GENE =</span> tgene[[x]], <span class="at">TERM2NAME =</span> tname[[x]]</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>    ))[, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="do">## G3</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>g3_sea <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(<span class="fu">seq_along</span>(tgene), <span class="cf">function</span>(x) {</span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="fu">enricher</span>(</span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>        g3, <span class="at">universe =</span> background, </span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">TERM2GENE =</span> tgene[[x]], <span class="at">TERM2NAME =</span> tname[[x]]</span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>    ))[, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine SEA results in a single data frame and export it as a .tsv file</span></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a><span class="do">## Combine data frames</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>sea_res <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a>    g1_sea <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"G1"</span>), </span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a>    g2_sea <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"G2"</span>),</span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>    g3_sea <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"G3"</span>)</span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a><span class="do">## Export .tsv</span></span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(</span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a>    sea_res,</span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"tables"</span>, <span class="st">"enrichment_bygroup.tsv"</span>)</span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The complete enrichment results are stored in the table <code>enrichment_bygroup.tsv</code>. To make visualization and interpretation easier, we will perform semantic similarity analysis to group redundant terms and get a global view of processes associated with each cluster.</p>
<p>Here, we will only use GO terms from the category “Biological Process”.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Semantic similarity analysis for GO-BP terms</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="do">## G1</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>g1_summary <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(<span class="fu">enricher</span>(</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    g1, <span class="at">universe =</span> background, </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">TERM2GENE =</span> go_gene, <span class="at">TERM2NAME =</span> go_des</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="do">## G2</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>g2_summary <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(<span class="fu">enricher</span>(</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>    g2, <span class="at">universe =</span> background, </span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">TERM2GENE =</span> go_gene, <span class="at">TERM2NAME =</span> go_des</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="do">## G3</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>g3_summary <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(<span class="fu">enricher</span>(</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>    g3, <span class="at">universe =</span> background, </span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">TERM2GENE =</span> go_gene, <span class="at">TERM2NAME =</span> go_des</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Save objects</span></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>    g1_summary, <span class="at">compress =</span> <span class="st">"xz"</span>,</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"result_files"</span>, <span class="st">"g1_summary.rda"</span>)</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>    g2_summary, <span class="at">compress =</span> <span class="st">"xz"</span>,</span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"result_files"</span>, <span class="st">"g2_summary.rda"</span>)</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a>    g3_summary, <span class="at">compress =</span> <span class="st">"xz"</span>,</span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"result_files"</span>, <span class="st">"g3_summary.rda"</span>)</span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, let’s plot the results.</p>
<div class="cell" data-fig.dpi="200">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tree plot</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>p_tree_g1 <span class="ot">&lt;-</span> <span class="fu">treeplot</span>(g1_summary, <span class="at">nWords =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>    ggsci<span class="sc">::</span><span class="fu">scale_fill_jama</span>() <span class="sc">+</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Group 1"</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>p_tree_g1<span class="sc">$</span>layers[[<span class="dv">4</span>]] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>p_tree_g2 <span class="ot">&lt;-</span> <span class="fu">treeplot</span>(g2_summary, <span class="at">nCluster =</span> <span class="dv">7</span>, <span class="at">nWords =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>    ggsci<span class="sc">::</span><span class="fu">scale_fill_jama</span>() <span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Group 2"</span>)</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>p_tree_g2<span class="sc">$</span>layers[[<span class="dv">4</span>]] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>p_tree_g3 <span class="ot">&lt;-</span> <span class="fu">treeplot</span>(g3_summary, <span class="at">nWords =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>    ggsci<span class="sc">::</span><span class="fu">scale_fill_jama</span>() <span class="sc">+</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Group 3"</span>)</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>p_tree_g3<span class="sc">$</span>layers[[<span class="dv">4</span>]] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace P.adj with -log10(P.adj)</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>p_tree_g1<span class="sc">$</span>data<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(p_tree_g1<span class="sc">$</span>data<span class="sc">$</span>color)</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>p_tree_g2<span class="sc">$</span>data<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(p_tree_g2<span class="sc">$</span>data<span class="sc">$</span>color)</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>p_tree_g3<span class="sc">$</span>data<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(p_tree_g3<span class="sc">$</span>data<span class="sc">$</span>color)</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots in one, with shared legends</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>rcol <span class="ot">&lt;-</span> <span class="fu">range</span>(</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>        p_tree_g1<span class="sc">$</span>data<span class="sc">$</span>color, p_tree_g2<span class="sc">$</span>data<span class="sc">$</span>color, p_tree_g3<span class="sc">$</span>data<span class="sc">$</span>color</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>rsize <span class="ot">&lt;-</span> <span class="fu">range</span>(</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>        p_tree_g1<span class="sc">$</span>data<span class="sc">$</span>count, p_tree_g2<span class="sc">$</span>data<span class="sc">$</span>count, p_tree_g2<span class="sc">$</span>data<span class="sc">$</span>count</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(p_tree_g1, p_tree_g2, p_tree_g3) <span class="sc">+</span></span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">"collect"</span>) <span class="sc">&amp;</span></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_continuous</span>(<span class="at">name =</span> <span class="st">"-Log10(P)"</span>, <span class="at">limits =</span> <span class="fu">signif</span>(rcol, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_continuous</span>(<span class="at">name =</span> <span class="st">"Gene count"</span>, <span class="at">limits =</span> rsize) <span class="sc">&amp;</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/plot-treeplot-1.png" class="img-fluid figure-img" width="1920"></p>
<figcaption class="figure-caption">Tree plot of functional terms associated with each orthogroup cluster.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell" data-fig.dpi="200">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dot plot</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>p_dot_g1 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(g1_summary, <span class="at">showCategory =</span> <span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Group 1"</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>p_dot_g2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(g2_summary, <span class="at">showCategory =</span> <span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Group 2"</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>p_dot_g3 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(g3_summary, <span class="at">showCategory =</span> <span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Group 3"</span>)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace P.adj with -log10(P.adj)</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>p_dot_g1<span class="sc">$</span>data<span class="sc">$</span>p.adjust <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(p_dot_g1<span class="sc">$</span>data<span class="sc">$</span>p.adjust)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>p_dot_g2<span class="sc">$</span>data<span class="sc">$</span>p.adjust <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(p_dot_g2<span class="sc">$</span>data<span class="sc">$</span>p.adjust)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>p_dot_g3<span class="sc">$</span>data<span class="sc">$</span>p.adjust <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(p_dot_g3<span class="sc">$</span>data<span class="sc">$</span>p.adjust)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots in one, keep shared legend</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>rcol <span class="ot">&lt;-</span> <span class="fu">range</span>(</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>        p_dot_g1<span class="sc">$</span>data<span class="sc">$</span>p.adjust, p_dot_g2<span class="sc">$</span>data<span class="sc">$</span>p.adjust, </span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>        p_dot_g3<span class="sc">$</span>data<span class="sc">$</span>p.adjust</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>rsize <span class="ot">&lt;-</span> <span class="fu">range</span>(</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>        p_dot_g1<span class="sc">$</span>data<span class="sc">$</span>Count, p_dot_g2<span class="sc">$</span>data<span class="sc">$</span>Count, p_dot_g3<span class="sc">$</span>data<span class="sc">$</span>Count</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(p_dot_g1, p_dot_g2, p_dot_g3) <span class="sc">+</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">"collect"</span>) <span class="sc">&amp;</span></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_continuous</span>(<span class="at">name =</span> <span class="st">"-Log10(P)"</span>, <span class="at">limits =</span> <span class="fu">signif</span>(rcol, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_continuous</span>(<span class="at">name =</span> <span class="st">"Gene count"</span>, <span class="at">limits =</span> rsize) <span class="sc">&amp;</span></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/plot-dotplot-1.png" class="img-fluid figure-img" width="1728"></p>
<figcaption class="figure-caption">Dotplot of functional terms associated with each orthogroup cluster.</figcaption>
</figure>
</div>
</div>
</div>
<p>The plots show that genes associated to particular biological processes tend to be clustered in the same orthogroup (group 3, scores closer to 1), while genes associated to other biological processes tend to be more dispersed across orthogroups (groups 1 and 2, scores closer to 1), possibly because they are evolving faster and, hence, have lower sequence similarity among themselves. In details, these genes and processes are:</p>
<ul>
<li><p><strong>Group 1:</strong> ATP production, water and K+ transport, seed oilbody biogenesis, and response to nitrate and ethylene.</p></li>
<li><p><strong>Group 2:</strong> sulfur amino acid metabolsm, spliceosome biogenesis, beta-1,3-glucan biosynthesis, response to brassinosteroids, xylem development, exocytosis, and calcium and sulfate transport.</p></li>
<li><p><strong>Group 3:</strong> photosynthesis, zinc and amino acid transport, DNA replication, endocytosis, cell-cell junction assembly, and toxin catabolism.</p></li>
</ul>
</section>
<section id="is-there-an-association-between-og-score-and-og-gene-length" class="level2" data-number="4.6">
<h2 data-number="4.6" class="anchored" data-anchor-id="is-there-an-association-between-og-score-and-og-gene-length"><span class="header-section-number">4.6</span> Is there an association between OG score and OG gene length?</h2>
<p><span class="citation" data-cites="emms2015orthofinder">Emms and Kelly (<a href="#ref-emms2015orthofinder" role="doc-biblioref">2015</a>)</span> have demonstrated a gene length bias that influences the accuracy of orthogroup detection. This is because short sequences cannot produce large bit scores or low e-values, and long sequences produce many hits with scores better than those for the best hits of short sequences <span class="citation" data-cites="emms2015orthofinder">(<a href="#ref-emms2015orthofinder" role="doc-biblioref">Emms and Kelly 2015</a>)</span>. OrthoFinder implements a score transform that claims to eliminate such bias. But does it remove the bias completely?</p>
<p>To answer this question, we will use homogeneity scores for the default OrthoFinder run (default DIAMOND mode, mcl = 1.5).</p>
<p>First of all, let’s calculate the mean and median gene length for each orthogroup.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine proteomes into a single AAStringSet object and clean gene names</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(brassicaceae_proteomes) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>proteomes <span class="ot">&lt;-</span> <span class="fu">do.call</span>(c, brassicaceae_proteomes)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(brassicaceae_proteomes)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(proteomes) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\\t</span><span class="st">.*"</span>, <span class="st">""</span>, <span class="fu">names</span>(proteomes))</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(proteomes) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, <span class="fu">names</span>(proteomes))</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(proteomes) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.[0-9]$"</span>, <span class="st">""</span>, <span class="fu">names</span>(proteomes))</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(proteomes) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.[0-9]</span><span class="sc">\\</span><span class="st">.p$"</span>, <span class="st">""</span>, <span class="fu">names</span>(proteomes))</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(proteomes) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.t[0-9]$"</span>, <span class="st">""</span>, <span class="fu">names</span>(proteomes))</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(proteomes) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.g$"</span>, <span class="st">""</span>, <span class="fu">names</span>(proteomes))</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Load only orthogroups from the default OrthoFinder run</span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>og <span class="ot">&lt;-</span> <span class="fu">read_orthogroups</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">"Orthogroups_default_1_5.tsv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Gene =</span> <span class="fu">str_replace_all</span>(</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>        Gene, <span class="fu">c</span>(</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\\\t</span><span class="st">.*"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\\</span><span class="st">.[0-9]$"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\\</span><span class="st">.[0-9]</span><span class="sc">\\</span><span class="st">.p$"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\\</span><span class="st">.t[0-9]$"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\\</span><span class="st">.g$"</span> <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Orthogroup, Gene)</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate mean gene lengths for each orthogroup</span></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>gene_lengths <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">Gene =</span> <span class="fu">names</span>(proteomes), </span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">Length =</span> Biostrings<span class="sc">::</span><span class="fu">width</span>(proteomes)</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>og_gene_lengths <span class="ot">&lt;-</span> og <span class="sc">%&gt;%</span></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(., gene_lengths) <span class="sc">%&gt;%</span> </span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Orthogroup) <span class="sc">%&gt;%</span></span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">mean_length =</span> <span class="fu">mean</span>(Length),</span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">median_length =</span> <span class="fu">median</span>(Length)</span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Add homogeneity scores to data frame of mean gene length per orthogroup</span></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>og_length_and_scores <span class="ot">&lt;-</span> og_assessment <span class="sc">%&gt;%</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(Mode <span class="sc">==</span> <span class="st">"default_1_5"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Orthogroups, Mean_H, Median_H) <span class="sc">%&gt;%</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(., og_gene_lengths, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Orthogroups"</span> <span class="ot">=</span> <span class="st">"Orthogroup"</span>))</span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Save data</span></span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>    og_length_and_scores,</span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"products"</span>, <span class="st">"result_files"</span>, <span class="st">"og_length_and_scores.rda"</span>),</span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">compress =</span> <span class="st">"xz"</span></span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, let’s explore a potential association between the two variables. Here, since the distribution of homogeneity scores is highly right-skewed, we will perform a correlation test using Spearman’s <span class="math inline">\(\rho\)</span> as a measure of association.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>ogl <span class="ot">&lt;-</span> og_length_and_scores<span class="sc">$</span>median_length</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>ogs <span class="ot">&lt;-</span> og_length_and_scores<span class="sc">$</span>Median_H</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate Spearman's correlation coefficient</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>cor_spearman <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(ogl, ogs, <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">exact =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>cor_spearman</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  ogl and ogs
S = 9.7041e+11, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
-0.190358 </code></pre>
</div>
</div>
<p>Finally, let’s plot the data and add the test statistics.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>p_association_length_homogeneity <span class="ot">&lt;-</span> og_length_and_scores <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">logH =</span> <span class="fu">log10</span>(Median_H <span class="sc">+</span> <span class="dv">1</span>),</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">logLength =</span> <span class="fu">log10</span>(median_length <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggscatter</span>(</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>        ., <span class="at">x =</span> <span class="st">"logLength"</span>, <span class="at">y =</span> <span class="st">"logH"</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>,</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"text"</span>,</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fl">1.7</span>, <span class="at">y =</span> <span class="fl">0.05</span>, </span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"rho"</span>, <span class="st">"=="</span>, <span class="fu">signif</span>(cor_spearman<span class="sc">$</span>estimate, <span class="dv">2</span>)),</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">parse =</span> <span class="cn">TRUE</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>        <span class="st">"text"</span>,</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fl">1.71</span>, <span class="at">y =</span> <span class="fl">0.03</span>,</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"P"</span>, <span class="st">"&lt;"</span>, <span class="fl">2.2e-16</span>), <span class="at">parse =</span> <span class="cn">TRUE</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Relationship between OG score and gene length"</span>,</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(Log[<span class="dv">10</span>] <span class="sc">~</span> <span class="st">"median gene length"</span>),</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(Log[<span class="dv">10</span>] <span class="sc">~</span> <span class="st">"median homogeneity score"</span>)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>p_association_length_homogeneity</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_orthogroup_inference_brassicaceae_files/figure-html/plot-og-length-h-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Relationship between sequence length and orthogroup scores.</figcaption>
</figure>
</div>
</div>
</div>
<p>The data and correlation test show that there is a significant negative correlation between homogeneity and OG gene length, but the correlation is weak (<span class="math inline">\(\rho = -0.19\)</span>). Although weak, this negative relationship shows that OrthoFinder can reduce the gene length bias, but not completely.</p>
</section>
<section id="session-info" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="session-info">Session info</h2>
<p>This document was created under the following conditions:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.3.0 (2023-04-21)
 os       Ubuntu 20.04.5 LTS
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Europe/Brussels
 date     2023-08-07
 pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package          * version   date (UTC) lib source
 abind              1.4-5     2016-07-21 [1] CRAN (R 4.3.0)
 AnnotationDbi      1.62.0    2023-04-25 [1] Bioconductor
 ape                5.7-1     2023-03-13 [1] CRAN (R 4.3.0)
 aplot              0.1.10    2023-03-08 [1] CRAN (R 4.3.0)
 backports          1.4.1     2021-12-13 [1] CRAN (R 4.3.0)
 beeswarm           0.4.0     2021-06-01 [1] CRAN (R 4.3.0)
 Biobase            2.60.0    2023-04-25 [1] Bioconductor
 BiocGenerics       0.46.0    2023-04-25 [1] Bioconductor
 BiocParallel       1.34.0    2023-04-25 [1] Bioconductor
 Biostrings         2.68.0    2023-04-25 [1] Bioconductor
 bit                4.0.5     2022-11-15 [1] CRAN (R 4.3.0)
 bit64              4.0.5     2020-08-30 [1] CRAN (R 4.3.0)
 bitops             1.0-7     2021-04-24 [1] CRAN (R 4.3.0)
 blob               1.2.4     2023-03-17 [1] CRAN (R 4.3.0)
 broom              1.0.4     2023-03-11 [1] CRAN (R 4.3.0)
 cachem             1.0.8     2023-05-01 [1] CRAN (R 4.3.0)
 car                3.1-2     2023-03-30 [1] CRAN (R 4.3.0)
 carData            3.0-5     2022-01-06 [1] CRAN (R 4.3.0)
 cli                3.6.1     2023-03-23 [1] CRAN (R 4.3.0)
 clusterProfiler  * 4.8.1     2023-05-03 [1] Bioconductor
 codetools          0.2-19    2023-02-01 [4] CRAN (R 4.2.2)
 cogeqc           * 1.4.0     2023-04-25 [1] Bioconductor
 coin               1.4-2     2021-10-08 [1] CRAN (R 4.3.0)
 colorspace         2.1-0     2023-01-23 [1] CRAN (R 4.3.0)
 commonmark         1.9.0     2023-03-17 [1] CRAN (R 4.3.0)
 cowplot            1.1.1     2020-12-30 [1] CRAN (R 4.3.0)
 crayon             1.5.2     2022-09-29 [1] CRAN (R 4.3.0)
 data.table         1.14.8    2023-02-17 [1] CRAN (R 4.3.0)
 DBI                1.1.3     2022-06-18 [1] CRAN (R 4.3.0)
 digest             0.6.33    2023-07-07 [1] CRAN (R 4.3.0)
 DOSE               3.26.1    2023-05-03 [1] Bioconductor
 downloader         0.4       2015-07-09 [1] CRAN (R 4.3.0)
 dplyr            * 1.1.2     2023-04-20 [1] CRAN (R 4.3.0)
 enrichplot       * 1.20.0    2023-04-25 [1] Bioconductor
 evaluate           0.21      2023-05-05 [1] CRAN (R 4.3.0)
 fansi              1.0.4     2023-01-22 [1] CRAN (R 4.3.0)
 farver             2.1.1     2022-07-06 [1] CRAN (R 4.3.0)
 fastmap            1.1.1     2023-02-24 [1] CRAN (R 4.3.0)
 fastmatch          1.1-3     2021-07-23 [1] CRAN (R 4.3.0)
 fgsea              1.26.0    2023-04-25 [1] Bioconductor
 forcats          * 1.0.0     2023-01-29 [1] CRAN (R 4.3.0)
 generics           0.1.3     2022-07-05 [1] CRAN (R 4.3.0)
 GenomeInfoDb       1.36.0    2023-04-25 [1] Bioconductor
 GenomeInfoDbData   1.2.10    2023-04-28 [1] Bioconductor
 ggbeeswarm         0.7.2     2023-04-29 [1] CRAN (R 4.3.0)
 ggforce            0.4.1     2022-10-04 [1] CRAN (R 4.3.0)
 ggfun              0.0.9     2022-11-21 [1] CRAN (R 4.3.0)
 ggnewscale         0.4.8     2022-10-06 [1] CRAN (R 4.3.0)
 ggplot2          * 3.4.1     2023-02-10 [1] CRAN (R 4.3.0)
 ggplotify          0.1.0     2021-09-02 [1] CRAN (R 4.3.0)
 ggpubr           * 0.6.0     2023-02-10 [1] CRAN (R 4.3.0)
 ggraph             2.1.0     2022-10-09 [1] CRAN (R 4.3.0)
 ggrepel            0.9.3     2023-02-03 [1] CRAN (R 4.3.0)
 ggsci              3.0.0     2023-03-08 [1] CRAN (R 4.3.0)
 ggsignif           0.6.4     2022-10-13 [1] CRAN (R 4.3.0)
 ggtext             0.1.2     2022-09-16 [1] CRAN (R 4.3.0)
 ggtree             3.8.0     2023-04-25 [1] Bioconductor
 glue               1.6.2     2022-02-24 [1] CRAN (R 4.3.0)
 GO.db              3.17.0    2023-05-02 [1] Bioconductor
 GOSemSim           2.26.0    2023-04-25 [1] Bioconductor
 graphlayouts       1.0.0     2023-05-01 [1] CRAN (R 4.3.0)
 gridExtra          2.3       2017-09-09 [1] CRAN (R 4.3.0)
 gridGraphics       0.5-1     2020-12-13 [1] CRAN (R 4.3.0)
 gridtext           0.1.5     2022-09-16 [1] CRAN (R 4.3.0)
 gson               0.1.0     2023-03-07 [1] CRAN (R 4.3.0)
 gtable             0.3.3     2023-03-21 [1] CRAN (R 4.3.0)
 HDO.db             0.99.1    2023-06-20 [1] Bioconductor
 here             * 1.0.1     2020-12-13 [1] CRAN (R 4.3.0)
 hms                1.1.3     2023-03-21 [1] CRAN (R 4.3.0)
 htmltools          0.5.5     2023-03-23 [1] CRAN (R 4.3.0)
 htmlwidgets        1.6.2     2023-03-17 [1] CRAN (R 4.3.0)
 httr               1.4.5     2023-02-24 [1] CRAN (R 4.3.0)
 igraph             1.4.2     2023-04-07 [1] CRAN (R 4.3.0)
 IRanges            2.34.0    2023-04-25 [1] Bioconductor
 jsonlite           1.8.7     2023-06-29 [1] CRAN (R 4.3.0)
 KEGGREST           1.40.0    2023-04-25 [1] Bioconductor
 knitr              1.43      2023-05-25 [1] CRAN (R 4.3.0)
 labeling           0.4.2     2020-10-20 [1] CRAN (R 4.3.0)
 lattice            0.20-45   2021-09-22 [4] CRAN (R 4.2.0)
 lazyeval           0.2.2     2019-03-15 [1] CRAN (R 4.3.0)
 libcoin            1.0-9     2021-09-27 [1] CRAN (R 4.3.0)
 lifecycle          1.0.3     2022-10-07 [1] CRAN (R 4.3.0)
 lubridate        * 1.9.2     2023-02-10 [1] CRAN (R 4.3.0)
 magrittr           2.0.3     2022-03-30 [1] CRAN (R 4.3.0)
 markdown           1.6       2023-04-07 [1] CRAN (R 4.3.0)
 MASS               7.3-58.2  2023-01-23 [4] CRAN (R 4.2.2)
 Matrix             1.5-1     2022-09-13 [4] CRAN (R 4.2.1)
 matrixStats        1.0.0     2023-06-02 [1] CRAN (R 4.3.0)
 memoise            2.0.1     2021-11-26 [1] CRAN (R 4.3.0)
 modeltools         0.2-23    2020-03-05 [1] CRAN (R 4.3.0)
 multcomp           1.4-25    2023-06-20 [1] CRAN (R 4.3.0)
 munsell            0.5.0     2018-06-12 [1] CRAN (R 4.3.0)
 mvtnorm            1.1-3     2021-10-08 [1] CRAN (R 4.3.0)
 nlme               3.1-162   2023-01-31 [4] CRAN (R 4.2.2)
 patchwork        * 1.1.2     2022-08-19 [1] CRAN (R 4.3.0)
 pillar             1.9.0     2023-03-22 [1] CRAN (R 4.3.0)
 pkgconfig          2.0.3     2019-09-22 [1] CRAN (R 4.3.0)
 plyr               1.8.8     2022-11-11 [1] CRAN (R 4.3.0)
 png                0.1-8     2022-11-29 [1] CRAN (R 4.3.0)
 polyclip           1.10-4    2022-10-20 [1] CRAN (R 4.3.0)
 purrr            * 1.0.1     2023-01-10 [1] CRAN (R 4.3.0)
 qvalue             2.32.0    2023-04-25 [1] Bioconductor
 R6                 2.5.1     2021-08-19 [1] CRAN (R 4.3.0)
 RColorBrewer       1.1-3     2022-04-03 [1] CRAN (R 4.3.0)
 Rcpp               1.0.10    2023-01-22 [1] CRAN (R 4.3.0)
 RCurl              1.98-1.12 2023-03-27 [1] CRAN (R 4.3.0)
 readr            * 2.1.4     2023-02-10 [1] CRAN (R 4.3.0)
 reshape2           1.4.4     2020-04-09 [1] CRAN (R 4.3.0)
 rlang              1.1.1     2023-04-28 [1] CRAN (R 4.3.0)
 rmarkdown          2.23      2023-07-01 [1] CRAN (R 4.3.0)
 rprojroot          2.0.3     2022-04-02 [1] CRAN (R 4.3.0)
 RSQLite            2.3.1     2023-04-03 [1] CRAN (R 4.3.0)
 rstatix          * 0.7.2     2023-02-01 [1] CRAN (R 4.3.0)
 rstudioapi         0.14      2022-08-22 [1] CRAN (R 4.3.0)
 S4Vectors          0.38.0    2023-04-25 [1] Bioconductor
 sandwich           3.0-2     2022-06-15 [1] CRAN (R 4.3.0)
 scales             1.2.1     2022-08-20 [1] CRAN (R 4.3.0)
 scatterpie         0.2.1     2023-06-07 [1] CRAN (R 4.3.0)
 sessioninfo        1.2.2     2021-12-06 [1] CRAN (R 4.3.0)
 shadowtext         0.1.2     2022-04-22 [1] CRAN (R 4.3.0)
 stringi            1.7.12    2023-01-11 [1] CRAN (R 4.3.0)
 stringr          * 1.5.0     2022-12-02 [1] CRAN (R 4.3.0)
 survival           3.5-3     2023-02-12 [4] CRAN (R 4.2.2)
 TH.data            1.1-2     2023-04-17 [1] CRAN (R 4.3.0)
 tibble           * 3.2.1     2023-03-20 [1] CRAN (R 4.3.0)
 tidygraph          1.2.3     2023-02-01 [1] CRAN (R 4.3.0)
 tidyr            * 1.3.0     2023-01-24 [1] CRAN (R 4.3.0)
 tidyselect         1.2.0     2022-10-10 [1] CRAN (R 4.3.0)
 tidytree           0.4.2     2022-12-18 [1] CRAN (R 4.3.0)
 tidyverse        * 2.0.0     2023-02-22 [1] CRAN (R 4.3.0)
 timechange         0.2.0     2023-01-11 [1] CRAN (R 4.3.0)
 treeio             1.24.1    2023-05-31 [1] Bioconductor
 tweenr             2.0.2     2022-09-06 [1] CRAN (R 4.3.0)
 tzdb               0.3.0     2022-03-28 [1] CRAN (R 4.3.0)
 utf8               1.2.3     2023-01-31 [1] CRAN (R 4.3.0)
 vctrs              0.6.3     2023-06-14 [1] CRAN (R 4.3.0)
 vipor              0.4.5     2017-03-22 [1] CRAN (R 4.3.0)
 viridis            0.6.2     2021-10-13 [1] CRAN (R 4.3.0)
 viridisLite        0.4.2     2023-05-02 [1] CRAN (R 4.3.0)
 withr              2.5.0     2022-03-03 [1] CRAN (R 4.3.0)
 xfun               0.39      2023-04-20 [1] CRAN (R 4.3.0)
 xml2               1.3.4     2023-04-27 [1] CRAN (R 4.3.0)
 XVector            0.40.0    2023-04-25 [1] Bioconductor
 yaml               2.3.7     2023-01-23 [1] CRAN (R 4.3.0)
 yulab.utils        0.0.6     2022-12-20 [1] CRAN (R 4.3.0)
 zlibbioc           1.46.0    2023-04-25 [1] Bioconductor
 zoo                1.8-12    2023-04-13 [1] CRAN (R 4.3.0)

 [1] /home/faalm/R/x86_64-pc-linux-gnu-library/4.3
 [2] /usr/local/lib/R/site-library
 [3] /usr/lib/R/site-library
 [4] /usr/lib/R/library

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>
</section>
<section id="references" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references">References</h2>


<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-emms2015orthofinder" class="csl-entry" role="listitem">
Emms, David M, and Steven Kelly. 2015. <span>“OrthoFinder: Solving Fundamental Biases in Whole Genome Comparisons Dramatically Improves Orthogroup Inference Accuracy.”</span> <em>Genome Biology</em> 16 (1): 1–14.
</div>
<div id="ref-emms2019orthofinder" class="csl-entry" role="listitem">
———. 2019. <span>“OrthoFinder: Phylogenetic Orthology Inference for Comparative Genomics.”</span> <em>Genome Biology</em> 20 (1): 1–14.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/03_orthogroup_inference_dbs.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Assessing orthogroup inference in public databases</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/05_synteny_assessment_fabaceae.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Assessing synteny detection in Fabaceae</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>